circuit/pallets/execution-delivery/src/lib.rs

```diff
@@ -883,91 +697,72 @@ impl<T: Config> Pallet<T> {
         contracts.extend(seen_contracts);
         contracts.extend(unseen_contracts);
 
-        let (current_block_no, block_zero) = (
+        let (_current_block_no, _block_zero) = (
             <frame_system::Pallet<T>>::block_number(),
             T::BlockNumber::zero(),
         );
 
-        let xtx_schedule = XtxSchedule::new_sequential_from_contracts(
-            contracts.clone(),
-            contract_ids.clone(),
-            composes.clone(),
-            action_descriptions.clone(),
-            None,
-            current_block_no.clone(),
-        )?;
+        Ok((contracts, contract_ids, action_descriptions))
+    }
 
-        let max_steps = contracts.len() as u32;
+    pub fn submit_xtx_execution(
```

Comment:
I feel like this needs a bit more discussion in general, as right now it:
* loads all necessary contracts to VVM (maybe that wont be necessary all the time)
* executes a pre_run (which internally runs every contract)
* creates a new Xtx record
* notifies that a new Xtx has been submitted

Maybe this also needs some further clean up as part of this revamp.

Reply:
https://github.com/t3rn/t3rn/pull/139#discussion_r742809564